src/ripple/app/ledger/LedgerEntrySet.cpp

```diff
@@ -1406,6 +1422,40 @@ TER LedgerEntrySet::trustDelete (
     return terResult;
 }
 
+void LedgerEntrySet::enableDeferredCredits (bool enable)
+{
+    if (!enable)
+    {
+        mDeferredCredits.reset ();
+        return;
+    }
+
+    if (!mDeferredCredits)
+        mDeferredCredits.emplace ();
+}
```

Comment:
`ScopedDeferCredits` is more about exception safety than nesting. I can see where you got that idea - that class does jump through some hoops to make sure it leave the `LedgerEntrySet` in the state it found it.

For now I'd like to leave the asserts out. The initial version of the code in payments would have hit the assert, and uses like that are legit. I'd rather use asserts for things that are definitely errors.